
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1234481874 18000
# Node ID c08c602ca0dcbc137a4ca965b3722953cabfa4fc
# Parent 4044df5a4d12fef4086d7e604444467eb105f348
"Maildir filename has wrong W value" error was missing new/cur directory in the middle.

--- a/src/lib-storage/index/maildir/maildir-mail.c	Tue Feb 10 12:19:05 2009 -0500
+++ b/src/lib-storage/index/maildir/maildir-mail.c	Thu Feb 12 18:37:54 2009 -0500
@@ -485,8 +485,12 @@ static void maildir_mail_set_cache_corru
 					       _mail->uid, &flags);
 		if (maildir_filename_get_size(fname, MAILDIR_EXTRA_VIRTUAL_SIZE,
 					      &size)) {
-			i_error("Maildir filename has wrong W value: %s/%s",
-				mbox->path, fname);
+			const char *subdir =
+				(flags & MAILDIR_UIDLIST_REC_FLAG_NEW_DIR) != 0 ?
+				"new" : "cur";
+			mail_storage_set_critical(_mail->box->storage,
+				"Maildir filename has wrong W value: %s/%s/%s",
+				mbox->path, subdir, fname);
 		} else if (maildir_uidlist_lookup_ext(mbox->uidlist, _mail->uid,
 				MAILDIR_UIDLIST_REC_EXT_VSIZE) != NULL) {
 			maildir_uidlist_set_ext(mbox->uidlist, _mail->uid,

